osu.Server.Spectator/Database/DatabaseAccess.cs

```diff
@@ -295,14 +295,15 @@ public async Task MarkScoreHasReplay(Score score)
             });
         }
 
-        public async Task<long?> GetScoreIdFromToken(long token)
+        public async Task<SoloScore?> GetScoreFromToken(long token)
         {
             var connection = await getConnectionAsync();
 
-            return await connection.QuerySingleOrDefaultAsync<long?>("SELECT `score_id` FROM `solo_score_tokens` WHERE `id` = @Id", new
-            {
-                Id = token
-            });
+            return await connection.QuerySingleOrDefaultAsync<SoloScore?>(
+                "SELECT * FROM `solo_scores` WHERE `id` = (SELECT `score_id` FROM `solo_score_tokens` WHERE `id` = @Id)", new
+                {
+                    Id = token
+                });
```

Comment:
I'll just note that this is changing the same method that my series of pulls concerning replay treatment is changing (see https://github.com/ppy/osu-server-spectator/pull/185).

This query is still - same as `master` - correct only for solo scores and wrong for multiplayer scores (see https://github.com/ppy/osu/pull/24697 / https://github.com/ppy/osu-infrastructure/blob/master/score-submission.md#multiplayer for full explanation), but I suppose it is no *more* wrong than `master`.

Make of that what you will wrt which pull should go first etc.

Reply:
I'm okay waiting, I don't see this as a higher priority than the other PR.

Reply:
disregard above, see https://github.com/ppy/osu-infrastructure/issues/24